### PR TITLE
actinia oberflaechenklassifikation - jupyter notebook

### PR DESCRIPTION
**Pull request by anikaweinmann**
Started the actinia introduction for the workshop.

<strike>TODO:</strike>
- <strike>@neteler fix raster visalization </strike> CHECK MN
-  <strike>übersetzen </strike> CHECK MN
- <strike>user vs. persistente DB</strike>

### CODE REVIEW

**Comment by neteler** [comment]
I have translated the entire notebook and updated several passages + fixed leafmap (credentials injection was missing).

**TODO**: add NDVI color table to COG in process chain

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 1]
```diff
@@ -0,0 +1,1383 @@
...
+   "cell_type": "markdown",
+   "id": "26859918",
+   "metadata": {},
+   "source": [
+    "<img style=\"float: center;\" src=\"https://raw.githubusercontent.com/actinia-org/actinia-core/main/docs/docs/actinia_logo.svg\" width=\"25%\">\n",
+    "\n",
+    "## Einführung\n",
+    "\n",
+    "Die cloudbasierte Geoprocessing-Plattform [actinia](https://actinia.mundialis.de/)\n",
+    "ist in der Lage, große Mengen von Geodaten in der Cloud zu verarbeiten und\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "ist in der Lage, große Mengen von Geodaten in der Cloud zu verarbeiten und zu\n",
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 2]
```diff
@@ -0,0 +1,1383 @@
...
+    "analysieren. Die Software vereinigt viele [GRASS GIS](https://grass.osgeo.org/)\n",
+    "Verarbeitungswerkzeuge und Datenbanken als\n",
+    "[REST-Dienst](https://en.wikipedia.org/wiki/Representational_State_Transfer).\n",
+    "Daher ist der Zugriff auf GRASS-Ressourcen wie Rasterkarten, Raum-Zeit-Rasterdatensätze,\n",
+    "Verarbeitungs- und Analysemodule sind über URLs verfügbar. Darüber hinaus ermöglicht\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "Verarbeitungs- und Analysemodule über URLs möglich. Darüber hinaus ermöglicht\n",
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 3]
```diff
@@ -0,0 +1,1383 @@
...
+    "actinia die cloudbasierte Verarbeitung von Daten, zum Beispiel alle Landsat 4-9 Szenen\n",
+    "sowie alle Sentinel-2-Szenen in einer ephemeren Datenbank. Die Berechnungsergebnisse\n",
+    "stehen in der ephemeren Verarbeitung über Objektspeicher als GeoTIFF/COG Rastedateien oder\n",
+    "GeoPackage-Vektordateinen zur Verfügung.\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "stehen in der ephemeren Verarbeitung über Objektspeicher als GeoTIFF/COG Rasterdateien oder\n",
    "GeoPackage-Vektordateien zur Verfügung.\n",
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 4]
```diff
@@ -0,0 +1,1383 @@
...
+    "\n",
+    "Der actinia-Dienst besteht aus dem *[actinia core](https://github.com/actinia-org/actinia-core)*,\n",
+    "der den grundlegenden, aber anspruchsvollen Verarbeitungsdienst bereitstellt, und\n",
+    "*[actinia plugins](https://github.com/orgs/actinia-org/repositories?q=actinia+plugins&type=all&language=&sort=)*,\n",
+    "die themenpezifische Dienste wie NDVI-Berechnungen aus Sentinel-2 oder Landsat-Daten,\n",
+    "räumlich-zeitliche statistische Analysen und vieles mehr.\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "räumlich-zeitliche statistische Analysen und vieles mehr anbieten.\n",
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 5]
```diff
@@ -0,0 +1,1383 @@
...
+    "\n",
+    "Das folgende Beispiel ist eine überarbeitete Jupyter Notebook Version des Online\n",
+    "[actinia tutorial](https://actinia-dev.mundialis.de/tutorial/introduction.html)."
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "[actinia Tutorials](https://actinia-dev.mundialis.de/tutorial/introduction.html)."
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 6]
```diff
@@ -0,0 +1,1383 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08c7f32c",
+   "metadata": {},
+   "source": [
+    "### Hilfsmodule und Funktionen\n",
+    "\n",
+    "<span style=\"color:red\">**TODO is this needed???**</span> --> MN: please no\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]: RESTman und JSON Formatter werden nur für das Original-Tutorial (nicht Jupyter Notebook) gebraucht, wo curl und/oder der browser benutzt wird --> löschen

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 7]
```diff
@@ -0,0 +1,1383 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f1a67b3a",
+   "metadata": {},
+   "source": [
+    "## Vorbereitung\n",
+    "\n",
+    "Im diese Sitzung im Jupyter Notebook zu initialisieren, laden wir einige Python Bibliotheken."
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "Um diese Sitzung im Jupyter Notebook zu initialisieren, laden wir einige Python Bibliotheken."
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 8]
```diff
@@ -0,0 +1,1383 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "451fdce3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# first, let's import the required packages.\n",
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "# Zuerst werden die erforderlichen Python Bibliotheken importiert.\n",
```

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 9]
```diff
@@ -0,0 +1,1383 @@
...
+    "\n",
+    "from pprint import pprint\n",
+    "import sys\n",
+    "import json\n",
+    "import shutil\n",
+    "import time\n",
+    "\n",
+    "import requests\n",
+    "from requests.auth import HTTPBasicAuth\n"
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]: Hier auch gleich leafmap?

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 10]
```diff
@@ -0,0 +1,1383 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c1479e11",
+   "metadata": {},
+   "source": [
+    "### Datenverwaltung\n",
+    "\n",
+    "Auflistung aller Orte, die über den Endpunkt `/locations` in der persistenten Datenbank von actinia verfügbar sind:"
```
Comment (metzm) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]:
```suggestion
    "Auflistung aller GRASS locations, die über den Endpunkt `/locations` in der persistenten Datenbank von actinia verfügbar sind:"
```

**Comment by metzm** [comment]
Sollen die outputs der einzelnen Zellen wirklich hier mit gespeichert werden?

**Comment by neteler** [comment]
Nein, das war nur zum einfachen Betrachten des Zwischenstandes.

**Comment by linakrisztian** [comment]
Added NDVI colorbar
TODO: visualize NDVI map (without leafmap)

File: actinia_oberflaechenklassifikation/actinia_fossgis.ipynb [hunk 11]
```diff
@@ -0,0 +1,1452 @@
...
+    "\n",
+    "\n",
+    "# import shutil\n",
+    "\n",
+    "# import requests\n",
+    "\n",
+    "# url = 'http://example.com/img.png'\n",
+    "# response = requests.get(url, stream=True)\n",
+    "# with open('img.png', 'wb') as out_file:\n",
+    "#     shutil.copyfileobj(response.raw, out_file)"
```
Comment (linakrisztian) [inline actinia_oberflaechenklassifikation/actinia_fossgis.ipynb]: Wird das hier noch benötigt?

**Comment by neteler** [comment]
NDVI colorbar formula for integer scaled NDVI
(TODO: `awk` formula needs to be adapted to current rescaling formula):

```
r.mapcalc "${COLPREFIX}.ndvi_int = round((float($NIR - $RED) / ($NIR + $RED) * 1.0) * 100)"
cat $GISBASE/etc/colors/ndvi | awk '{printf "%02d %02d %02d %02d\n", $1*100, $2, $3, $4}' > /tmp/ndvi_100.col
r.colors map=${COLPREFIX}.ndvi_int rules=/tmp/ndvi_100.col
```

**Comment by metzm** [comment]
> NDVI colorbar formula for integer scaled NDVI (TODO: `awk` formula needs to be adapted to current rescaling formula):
> 
> ```
> r.mapcalc "${COLPREFIX}.ndvi_int = round((float($NIR - $RED) / ($NIR + $RED) * 1.0) * 100)"
> cat $GISBASE/etc/colors/ndvi | awk '{printf "%02d %02d %02d %02d\n", $1*100, $2, $3, $4}' > /tmp/ndvi_100.col
> r.colors map=${COLPREFIX}.ndvi_int rules=/tmp/ndvi_100.col
> ```

A simpler alternative is using `r.colors color=ndvi scale=100`

**Review by metzm — APPROVED** [review]
I suggest to merge, also the other PR, and then reorganize the files, e.g. notebooks next to each other, one common folder for images.

**Comment by neteler** [comment]
> A simpler alternative is using `r.colors color=ndvi scale=100`

Also, when we integer-scale NDVI with this formula? 

`r.mapcalc "ndvi_INT = round(127.5 * (1.0 + float($NIR - $RED) / float($NIR + $RED)))"`

I expected that the formula needs a change then.

**Comment by linakrisztian** [comment]
> I suggest to merge, also the other PR, and then reorganize the files, e.g. notebooks next to each other, one common folder for images.

:+1: I will merge, and will add better colorbars tomorrow.

**Pull request merged**